internal/services/compute/managed_disk_resource_test.go

```diff
@@ -758,6 +773,110 @@ resource "azurerm_managed_disk" "test" {
 `, data.Locations.Primary, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
 }
 
+func (r ManagedDiskResource) galleryImageDataDisk(data acceptance.TestData) string {
```

Comment:
This test does below things:
1. Create a linux vm, with the dependency template from LinuxVirtualMachineResource{}
2. Create a data disk attachment to the vm
3. Create an image gallery, an image definition and an image version from the vm with data disk attached
4. Create the disk out of the data disk of the image in the gallery